snapcraft/parts/lifecycle.py

```diff
@@ -556,6 +559,13 @@ def _run_in_provider(
             providers.capture_logs_from_instance(instance)
 
 
+def _expose_prime(project_path: Path, instance: Executor):
+    """Expose the instance's prime directory in ``project_path`` on the host."""
+    host_prime = project_path / "prime"
+    host_prime.mkdir(exist_ok=True)
+    instance.mount(host_source=project_path / "prime", target=Path("/root/prime"))
```

Comment:
We can use `utils.get_managed_environment_home_path()` instead of hard-coding `/root` (to prevent problems in case we change the work dirs location inside the instance), and obtain the prime directory name by creating a `craft_parts.ProjectDirs` object passing the managed environment home path as the work dir.